bin/proving-service/src/api/prover.rs

```diff
@@ -5,15 +5,49 @@ use miden_objects::{
 };
 use miden_tx::{LocalTransactionProver, TransactionProver};
 use miden_tx_batch_prover::LocalBatchProver;
+use serde::{Deserialize, Serialize};
 use tokio::sync::Mutex;
 use tonic::{Request, Response, Status};
 use tracing::instrument;
 
-use crate::{
-    commands::worker::ProverType,
-    generated::{ProofType, ProvingRequest, ProvingResponse, api_server::Api as ProverApi},
-    utils::MIDEN_PROVING_SERVICE,
-};
+use crate::generated::{self, ProvingRequest, ProvingResponse, api_server::Api as ProverApi};
+
+pub const MIDEN_PROVING_SERVICE: &str = "miden-proving-service";
+
+/// Specifies the type of proving supported.
+#[derive(Debug, Clone, Copy, Default, PartialEq, Serialize, Deserialize)]
+pub enum ProofType {
+    /// Transaction proving
```

Comment:
nit: Unnecessary comments maybe. Also tend to put a full stop at the end of the sentence.